### ISSUE TITLE
Discard changes not working as intended with slug fields

### ISSUE THREAD

**Issue by saasen** [sanity-io/sanity#430]
We are experiencing an issue when discarding changes after changing a field in a document which also has a slug field which is sourced to that field is discarded. 

Versions:
```
// package.json
"@sanity/base": "^0.123.0",
"@sanity/components": "^0.123.0",
"@sanity/core": "^0.123.0",
"@sanity/default-layout": "^0.123.0",
"@sanity/default-login": "^0.121.1",
"@sanity/desk-tool": "^0.123.0",
"react": "^15.6",
"react-dom": "^15.6",
"sanity-plugin-vision": "^2.0.7"

Node: 8.4.0
NPM: 5.5.1
```

Example schema:

```
// movie.js
export default {
    title: 'Movie',
    name: 'movie',
    type: 'document',
    fields: [
        {
            title: 'Name',
            name: 'name',
            type: 'string'
        },
        {
            title: 'Slug',
            name: 'slug',
            type: 'slug',
            options: {
                source: 'name'
            }
        }
    ]
}
```

Reproducing:
* Create an instance of the content model
* Give the `name` a name, the slug will be auto-generated
* Publish
* Add a word to the title (the slug will follow)
* Discard the changes
* The 'Publish changes' button is still active, and the Vision plug-in shows strange results:

![image](https://user-images.githubusercontent.com/3972118/34054424-385a3e84-e1cb-11e7-87a0-6ed76f0df9bf.png)

When experiencing this issue the first time, we had different results, but the result is still strange. We are expecting the document to get properly discarded an the previous version published.

**Comment by rexxars** [comment]
Thanks for a thorough issue report! We'll investigate and get back to you.

**Comment by evenwestvang** [comment]
Should be fixed in #585!
